docs: Update redirects i18n documentation for App Router (#81347)
vercel/next.js: docs/01-app/03-api-reference/05-config/01-next-config-js/redirects.mdx

<AppOnly>

> When implementing redirects with internationalization in the App Router, you handle locales through [dynamic route segments and middleware](/docs/app/guides/internationalization). This means you need to include locale information directly in your redirect paths:

Reviewer: ```suggestion
When implementing redirects with internationalization in the App Router, handle locales using [dynamic route segments and middleware](/docs/app/guides/internationalization). This means you must include the locale directly in your redirect paths:
```

</PagesOnly>

<AppOnly>

> ```js filename="next.config.js"

Reviewer: I've made some commit tweaks, but I wonder if we should align with our example, which uses middleware for this:

- https://github.com/vercel/next.js/blob/canary/examples/i18n-routing/middleware.ts

We should at least say, that the middleware approach can make use of the preferred locale.

Maybe:

```
When implementing redirects with internationalization in the App Router, you can include locales in `next.config.js` redirects, but only as hardcoded paths.

For dynamic or per-request locale handling, use [dynamic route segments and middleware](/docs/app/guides/internationalization), which can redirect based on the user’s preferred language.
```

I'll think about this a bit more too

--- 

I think we can also re-order this as:

````mdx
<AppOnly>
\
When implementing redirects with internationalization in the App Router, handle locales using [dynamic route segments and middleware](/docs/app/guides/internationalization). 

```js filename="next.config.js"
// config here
```
 
</AppOnly>

<PagesOnly>

When leveraging...

```js filename="next.config.js"
module.exports = {
  i18n: {
    locales: ['en', 'fr', 'de'],
    defaultLocale: 'en',
  },
// rest of config
```

</PagesOnly>

In some rare cases, you might...
````

}
```

</PagesOnly>

> In some rare cases, you might need to assign a custom status code for older HTTP Clients to properly redirect. In these cases, you can use the `statusCode` property instead of the `permanent` property, but not both. To to ensure IE11 compatibility, a `Refresh` header is automatically added for the 308 status code.

Reviewer: ```suggestion
In some rare cases, you might need to assign a custom status code for older HTTP Clients to properly redirect. In these cases, you can use the `statusCode` property instead of the `permanent` property, but not both. To ensure IE11 compatibility, a `Refresh` header is automatically added for the 308 status code.
```
